x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/TokensInvalidationResult.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    public static TokensInvalidationResult emptyResultWithCounter(int attemptCount) {
+        return new TokensInvalidationResult(Collections.emptyList(), Collections.emptyList(), Collections.emptyList(), attemptCount);
+    }
+
+    public List<String> getInvalidatedTokens() {
+        return invalidatedTokens;
+    }
+
+    public List<String> getPreviouslyInvalidatedTokens() {
+        return previouslyInvalidatedTokens;
+    }
+
+    public List<ElasticsearchException> getErrors() {
+        return errors;
+    }
+
+    public int getAttemptCount() {
+        return attemptCount;
+    }
+
+    public static void writeTo(TokensInvalidationResult result, StreamOutput out) throws IOException {
+        out.writeStringList(result.getInvalidatedTokens());
+        out.writeStringList(result.getPreviouslyInvalidatedTokens());
+        out.writeVInt(result.getErrors().size());
+        for (Exception e : result.getErrors()) {
+            out.writeException(e);
+        }
+        out.writeVInt(result.getAttemptCount());
+    }
+
+    public static TokensInvalidationResult readFrom(StreamInput in) throws IOException {
+        List<String> invalidatedTokens = in.readList(StreamInput::readString);
+        List<String> previouslyUnvalidatedTokens = in.readList(StreamInput::readString);
+        int errorsSize = in.readVInt();
+        List<ElasticsearchException> errors = new ArrayList<>(errorsSize);
+        for (int i = 0; i < errorsSize; i++) {
+            errors.add(in.readException());
+        }
+        int attemptCounter = in.readVInt();
+        return new TokensInvalidationResult(invalidatedTokens, previouslyUnvalidatedTokens, errors, attemptCounter);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject()
+            .field("invalidated_tokens", invalidatedTokens.size())
+            .field("previously_invalidated_tokens", previouslyInvalidatedTokens.size())
+            .field("error_size", errors.size());
+        if (errors.isEmpty() == false) {
+            builder.field("error_details");
+            builder.startArray();
+            for (ElasticsearchException e : errors) {
+                builder.startObject();
+                ElasticsearchException.generateThrowableXContent(builder, params, e);
+                builder.endObject();
+            }
+            builder.endArray();
+        }
+        return builder.endObject();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        TokensInvalidationResult that = (TokensInvalidationResult) o;
+        if (errors.size() != that.errors.size()) {
+            return false;
+        }
+        for (ElasticsearchException e : errors) {
```

Comment:
Not entirely sure how/if `equals` should be handled. This one seems like a fair assumption for equality but on the client side, because of `toXContent` and `fromXContent` transitions adding details, I couldn't come up with a proper way to consistently check for equivalence (https://github.com/elastic/elasticsearch/pull/35388/files#diff-a761e32018338ed3001a69b239d374b2R89). Suggestions are more than welcome.

Reply:
I don't know that you really need an `equals` method do you? Where is it going to be used?